robot/client/client.go

```diff
@@ -71,7 +71,7 @@ type RobotClient struct {
 	dialOptions []rpc.DialOption
 
 	mu                       sync.RWMutex
-	resourceNames            []resource.Name
+	cachedMachineStatus      robot.MachineStatus
```

Comment:
Should this be `*robot.MachineStatus` to indicate a possible state where we haven't cached anything? Would mean you didn't have to return `robot.MachineStatus{}` in some of your error cases and could just return `nil`.

Reply:
> Should this be *robot.MachineStatus to indicate a possible state where we haven't cached anything?

agreed - it would be helpful to at least make the internal `cachedMachineStatus` field a pointer to better express the case where nothing was cached yet. 

however...

> Would mean you didn't have to return robot.MachineStatus{} in some of your error cases and could just return nil.

the [MachineStatus interface method specifies a struct return value instead of a struct pointer](https://github.com/viamrobotics/rdk/blob/59e3c5c4beefdbcee91c9d959ed4871d9d601169/robot/robot.go#L162-L163), so that method would still have to return `MachineStatus{}` values. we can revisit this interface typing decision if folks are strongly against it. I wanted to avoid having to do worry about excessive `nil` checking on `MachineStatus`, but depending on how large that structure grows it might be useful to make it a pointer for performance reasons.

Reply:
> make it a pointer for performance reasons

discussed offline with @dgottlieb - the `resource.Status` objects, whose fields are most likely to expand in the short-term, are in slice and therefore stored on the heap and don't create extra work when copying.

there's no immediate plans to add any fields to the top-level struct - perhaps we can revisit this interface type if/when that changes

Reply:
Yeah fair; thanks for the response here! Leaving as-is seems fine to me given your points.

Reply:
cool, as-is for both the internal `cachedMachineStatus` field as well as the public method?

Reply:
I don't feel too strongly. I guess I noticed everything was `robot.MachineStatus` and not `*robot.MachineStatus`, and felt it to be odd. Given your reasoning for keeping the public method as `robot.MachineStatus`, I think either value or pointer is fine for the cached value.